src/java/map/FloatMask.java

```diff
@@ -306,6 +324,58 @@ public FloatMask clampMax(float val) {
         return this;
     }
 
+    public FloatMask enlarge(int size) {
+        float[][] largeMask = new float[size][size];
+        int smallX;
+        int smallY;
+        for (int x = 0; x < size; x++) {
+            smallX = StrictMath.min(x / (size / getSize()), getSize() - 1);
+            for (int y = 0; y < size; y++) {
+                smallY = StrictMath.min(y / (size / getSize()), getSize() - 1);
+                largeMask[x][y] = get(smallX, smallY);
+            }
+        }
+        mask = largeMask;
+        applySymmetry(symmetryHierarchy.getSpawnSymmetry());
+        VisualDebugger.visualizeMask(this);
+        return this;
+    }
+
+    public FloatMask enlarge2 (int size) {
+        float[][] largeMask = new float[size][size];
+        int oldSize = getSize();
+        for (int x = 0; x < size; x++) {
+            for (int y = 0; y < size; y++) {
+                largeMask[x][y] = get(Math.round(x / (size / oldSize)), Math.round(y / (size / oldSize)));
```

Comment:
Use StrictMath.round

Reply:
Also these are all integer division so no rounding is needed

Reply:
Need to check if the values are in bounds for the mask

Reply:
The x does not need to be calculated in the inner loop this will be less efficient

Reply:
Also enlarge2 is the same function as enlarge why does it exist?